### PR TITLE
Add monitor terms to glossary

### PR DESCRIPTION
**Pull request by estherk15**
### What does this PR do? What is the motivation?

- Requested by Product team to add terms listed in Technical solutions glossary.
- Added the following:
    - alerting type
    - anomaly
    - change alert
    - conditional variables
    - evaluation frequency
    - outlier
    - simple alert
    - threshold alert
- [DOCS-5667](https://datadoghq.atlassian.net/browse/DOCS-5667)

### Merge instructions

- [x] Please merge after reviewing

[DOCS-5667]: https://datadoghq.atlassian.net/browse/DOCS-5667?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: archetypes/glossary.md [hunk 1]
```diff
@@ -12,4 +16,4 @@ related_terms:
     - related_term1
     - related_term2
 ---
-Add glossary term definition and markdown here.
\ No newline at end of file
+Add glossary term definition and markdown here. For more information, <a href="example/with/link">see the documentation</a>.
```
Comment (apigirl) [inline archetypes/glossary.md:19]: I think this is what is breaking the link checker. May just want to describe that you need to use an HTML link (or change it to link to something real, like https://www.w3schools.com/html/html_links.asp ?)

File: content/en/glossary/terms/alerting_type.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+title: alerting type
+core_product:
+    - alerts
+synonyms:
+    - alert grouping
+related_terms:
+    - simple alert
+    - multi alert
+---
+Alerts are grouped automatically based on your selection of the `group by` aggregation step when defining your metric. Any added aggregations results in a "multi alert" type and no aggregation is a "simple alert" type. For more information, <a href="/monitors/configuration/?tab=thresholdalert#alert-grouping">see the documentation</a>.
```
Comment (apigirl) [inline content/en/glossary/terms/alerting_type.md:11]:
```suggestion
Alerts are grouped automatically based on your selection of the `group by` aggregation step when defining your metric. Any added aggregations result in a "multi alert" type, and no aggregation is a "simple alert" type. For more information, <a href="/monitors/configuration/?tab=thresholdalert#alert-grouping">see the documentation</a>.
```

File: content/en/glossary/terms/anomaly.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+title: anomaly
+core_product:
+    - alerts
+    - metrics
+related_terms:
+    - outlier
+---
+Anomaly detection is an algorithmic feature that identifies when a metric is behaving differently than it has in the past. It takes into account trends, seasonal day-of-week, and time-of-day patterns. It is suited for metrics with strong trends and recurring patterns that are hard to monitor with threshold-based alerting. For more information, <a href="/monitors/types/anomaly/">see the documentation</a>.
```
Comment (apigirl) [inline content/en/glossary/terms/anomaly.md:9]:
```suggestion
Anomaly detection is an algorithmic feature that identifies when a metric behaves differently than it has in the past. It takes into account trends, seasonal day-of-week, and time-of-day patterns. It is suited for metrics with strong trends and recurring patterns that are hard to monitor with threshold-based alerting. For more information, <a href="/monitors/types/anomaly/">see the documentation</a>.
```

File: content/en/glossary/terms/outlier.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+title: outlier
+core_product:
+    - alerts
+related_terms:
+    - anomaly
+---
+Outlier detection is an algorithmic feature that allows you to detect when a specific metric group is behaving different compared to its peers. For more information, <a href="/monitors/types/outlier">see the documentation</a>.
```
Comment (apigirl) [inline content/en/glossary/terms/outlier.md:8]:
```suggestion
Outlier detection is an algorithmic feature that allows you to detect when a specific metric group behaves differently than its peers. For more information, <a href="/monitors/types/outlier">see the documentation</a>.
```

File: content/en/glossary/terms/threshold_alert.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+title: threshold alert
+core_product:
+    - alerts
+related_terms:
+    - change alert
+---
+Monitor detection method that compares metric values to a static threshold. This is automatically selected as the default detection method for metric monitors. For more information, <a href="/monitors/types/metric/?tab=threshold#choose-the-detection-method">see the documentation</a>.
```
Comment (apigirl) [inline content/en/glossary/terms/threshold_alert.md:8]:
```suggestion
Monitor detection method that compares metric values to a static threshold. Threshold alerts are the default detection method for metric monitors. For more information, <a href="/monitors/types/metric/?tab=threshold#choose-the-detection-method">see the documentation</a>.
```

**Review by apigirl — COMMENTED** [review]
Some nits and also please fix the broken link checker

File: archetypes/glossary.md (hunk 1 above)
Comment (estherk15) [reply archetypes/glossary.md:19]: Ah how embarrassing :)

Comment (estherk15) [reply archetypes/glossary.md:19]:
```suggestion
Add glossary term definition and markdown here. 
```

File: content/en/glossary/terms/conditional_variables.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+title: conditional variables
+core_product:
+    - alerts
+---
+Conditional variables use if-else logic to display a different message depending on the state of the monitor and the details of how it was triggered. For more information, <a href="monitors/notify/variables/?tab=is_alert#conditional-variables">see the documentation</a>.
```
Comment (estherk15) [inline content/en/glossary/terms/conditional_variables.md:6]:
```suggestion
Conditional variables use if-else logic to display a different message depending on the state of the monitor and the details of how it was triggered. For more information, <a href="/monitors/notify/variables/?tab=is_alert#conditional-variables">see the documentation</a>.
```

**Review by apigirl — APPROVED** [review]

**Pull request merged**